File: Leetcode/680-Valid-Palindrome/ValidPalindromeII.go
func validPalindrome(s string) bool {
	return valid(s, 0, len(s)-1, 0)
}

func valid(s string, left int, right int, times int) bool {
	for left < right {
		if s[left] != s[right] {
			if times < 1 {
				return valid(s, left+1, right, times+1) || valid(s, left, right-1, times+1)
			} else {
				return false
			}
		}
		left++
		right--
	}
	return true
}

File: Leetcode/680-Valid-Palindrome/ValidPalindromeII.py
class Solution(object):
    def validPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        return self.valid(s, 0, len(s)-1, 0)
    
    def valid(self, s, left, right, times):
        """
        :type s: str
        :type left: int
        :type right: int
        :type times: int
        :rtype: bool
        """
        while left < right :
            if s[left] != s[right] :
                if times < 1:
                    return self.valid(s, left+1, right, times+1) or self.valid(s, left, right-1, times+1)
                return False
            left+=1
            right-=1
        return True


